File: euler50.py
from primeGen import sieve

primes_hash = sieve(1000000)
primes = [x+1 for x in range(len(primes_hash)) if primes_hash[x]==1]

max_len = 1
prime = 2

for x in range(len(primes)):
    smm = primes[x]
    cnt = x+1
    if primes[x]+primes[x+1] > 1000000:
        break
    while smm<1000000:
        smm+=primes[cnt]
        if smm>1000000:
            break
        if primes_hash[smm-1]:
            length = cnt-x+1
            if length>max_len:
                max_len=length
                prime = smm
        cnt+=1
print max_len, prime

File: euler50.cpp
// consecutive prime sum

#include <iostream>
#include <cmath>
using namespace std;

int primes[500000]={0};
int total;

void primegen()
{
	primes[0]=2;
	int cnt=1;
	for(int i=3; i<=1000000; i+=2){
		bool flag=true;
		for(int j=0; primes[j]<=(int)sqrt(i)+1 and primes[j]!=0;j++){
			if(i%primes[j]==0){
				flag=false;
				break;
			}
		}
		if(flag==true){primes[cnt]=i;
		cnt++;
	}
	}
	::total=cnt;
}

bool inarr(int *arr, int len, int val)
{
	for (int i=0;i<len;i++) {
		if(arr[i]==val)return true;
		if(arr[i]>val) return false;
	}
	return false;
}

int main()
{
	primegen();
	int *counts = new int[::total];
	int index;
	int flag=1;
	int maxcnt=2;
	for (int i=total-1; i>0; i--) {

		for(int j=0; primes[j]<=primes[i]/maxcnt; j++) {

			int sum=0, temp=j, cnt=0;

			while(1) {
				if(sum>primes[i]){break;}


				sum+=primes[temp++];
				cnt++;

				if(sum==primes[i]){
					if(cnt>maxcnt) {
						maxcnt=cnt;
						index=i;
					}
					break;
					flag=0;
				}
			}
			if(flag==0)break;
		}
	}
	cout << endl<<primes[index];
	cout << endl<< maxcnt<<endl;
}
